src/App.test.jsx

```diff
@@ -0,0 +1,23 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { MemoryRouter } from 'react-router-dom';
+
+import App from './App';
+
+test('renders loading', () => {
+  render(<App />);
+  const loading = screen.getByText(/hold on.../i);
+  expect(loading).toBeInTheDocument();
+});
+
+test('renders detail when clicking windy', async () => {
+  render(
+    <MemoryRouter>
```

Comment:
App shouldn't need the `MemoryRouter` as routes already exist on your top level component